WooCommerce/Classes/ViewRelated/Dashboard/ProductStock/ProductStockDashboardCardViewModel.swift

```diff
@@ -44,12 +48,29 @@ final class ProductStockDashboardCardViewModel: ObservableObject {
 
     func updateStockType(_ type: StockType) {
         selectedStockType = type
+        stores.dispatch(AppSettingsAction.setLastSelectedStockType(siteID: siteID, type: type.rawValue))
         Task {
             await reloadData()
         }
     }
 }
 
+private extension ProductStockDashboardCardViewModel {
+    @MainActor
+    func loadLastSelectedStockType() async -> StockType {
+        await withCheckedContinuation { continuation in
+            stores.dispatch(AppSettingsAction.loadLastSelectedStockType(siteID: siteID, onCompletion: { type in
+                guard let type else {
+                    continuation.resume(returning: .lowStock)
+                    return
+                }
+                let stockType = StockType(rawValue: type) ?? .lowStock
+                continuation.resume(returning: stockType)
```

Comment:
Super nit: We could unwrap `stocktype` in the `guard` statement to avoid falling back to `.lowStock` at two places. 

```suggestion
                guard let type,
                      let stockType = StockType(rawValue: type) else {
                    continuation.resume(returning: .lowStock)
                    return
                }
                continuation.resume(returning: stockType)
```